### ISSUE TITLE
wrap_vfp_fini needs null guard around drop

### ISSUE THREAD

**Issue by dward** [gquintard/varnish-rs#20]
wrap_vfp_fini appears to need a null check around the drop of priv1 as it may remain null during a failure or an explicit pass during init. The drop method panics when encountering a null pointer.

I'm able to reproduce the crash by modifying examples/vmod_vfp/src/lib.rs to return InitResult::Pass from the new method:

```
fn new(_: &mut Ctx, _: &mut VFPCtx) -> InitResult<Self> {
    InitResult::Pass
}
```

and by modifying the struct to contain alloc'd data:
```
struct Lower {
    buffer: Vec<u8>,
}
```

**Comment by gquintard** [comment]
Good catch, thank you Dustin. Can ask you for a VTC in the PR?
You might need to create a new function in the test vmods, but that's
definitely a check worth having.

**Comment by dward** [comment]
@gquintard - Sure thing. I'll update the PR once I create a VTC.

**Comment by gquintard** [comment]
Thanks! Just for information, I'm away from computers until may 27th, so
expect lag, but I'm definitely merging this as soon as I can

**Comment by dward** [comment]
@gquintard - Sounds good! I've updated the pull with the test.

**Comment by gquintard** [comment]
fixed by #21
